src/provider.c

```diff
@@ -216,17 +218,17 @@ static int provider_get_params(void *provctx, OSSL_PARAM params[])
             if (params->data_type != OSSL_PARAM_UTF8_PTR)
                 return 0;
 
-	    params->data = ctx->provider_name;
+            *((char **)params->data) = ctx->provider_name;
```

Comment:
Alternatively one could use the he OSSL_PARAM_set utility functions.

Reply:
hmm those are functions (not macros) from params.h, if i use them i have to link the provider with libcrypto. Id prefer not doing that.

Reply:
Maybe it would make sense that the core passes those via init functions "in" dispatch table.

Reply:
That's my opinion as well, I expect most provider authors to need to either write their own wrappers or to link against those functions.

Reply:
I ll go with own wrappers for now and not call any openssl functions other than the ones provided by the core. I believe not doing so leads to hard-to-debug problems:  already tested the master branch's rsa keygen with opencryptoki softtoken (which uses libcrypto) and keygen failed due to some problem with random number generation. If i use a HW token (no openssl dep.), it works fine.